### ISSUE TITLE
A11y_Model Builder_Create New Compute_Info and Relationships: Visual representation (*) is not provided for mandatory fields in the popup.

### ISSUE THREAD

**Issue by NiharikaTipirishetty** [dotnet/machinelearning-modelbuilder#1579]
[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.

**GitHubTags:**
#A11y_Model Builder; #A11yMAS; #A11yTCS; #WCAG1.3.1; #Info and Relationships; #Win32;
#DesktopApp; #A11ySev2; #ModelBuilder-Win32-Jun2021; #Visual Studio Client;

### Environment Details:
Application name: Model Builder

### OS Details:
Windows Version 21H1 OS Build (19043.1083)
Microsoft Visual Studio

### Note:
This issue is observed in all popups. In 'Create new workspace', 'Create new resource group' popups  visual representation for edit fields is not provided

### Repro Steps:

1. Enable ML.NET Model builder as a preview feature in tools>options>Environment>preview features.
2. Create a .NET C# project and tab till project under solution explorer and Add machine learning.
3. Tab till Image classification under scenario tab and press enter.
4. Select Azure under environment tab and tab till set up workspace and press enter.
5. In 'Create New Experiment' dialog, tab till 'New' link beside create new compute name field and press enter
6. Visual representation (*) is not provided for mandatory fields in create new compute popup.

### Actual Result:
Visual representation (*) is not provided for mandatory fields in create new compute popup.

### Expected Result:
Visual representation (*) should be provided for mandatory fields in create new compute popup.

### User Impact:
Screen Reader users will not know that a field is required if field is not given a visual representation

**Comment by beccamc** [comment]
We may need to pull in Rohan here for how we want to display mandatory fields. Should they be highlighted in red when empty, similar to the VS New project dialogs?

**Comment by beccamc** [comment]
Rohan, we haven't used the * to denote a required field before. We do have a red outline behavior for missing fields on some of the  Azure fields.

**Comment by beccamc** [comment]
@romalpani Can you take a look here?

**Comment by romalpani** [comment]
@beccamc 
To make the field "required", we don't use an asterisk in VS (yet). The patterns that we do use are shown here in the guidelines: https://devdivdesignguide.azurewebsites.net/windows/notifications-and-progress/errors-and-validation/#field-validation

For the problem with the "Create new experiment" dialog, the "Compute name" text field should have the "<Required>" watermark in it. And supported with validation in case a customer leaves the field empty. The dev-tools team that works on connected services already uses a control that offers this kind of validation (screenshot below). It may save us time if we can reuse that control. 

What are the rules for a valid Compute name? We can create an error message (something as seen in the screenshot below).

![image](https://user-images.githubusercontent.com/48927690/130147284-b05906ac-24e6-495e-bfbf-f5eca4a060ac.png)

**Comment by beccamc** [comment]
I got some info from Phil about their error handling control 

Ok.  I think you can take a look at: https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=%2Fdocs%2FShared%2FValidatableControls%2FValidatableControls.md&_a=contents&version=GBmain to get a high level read on how we do this at a control level

ValidatableInputField.xaml does a lot of the work

And a common use looks like: 
<validatable:ValidatableInputField            Grid.Row="1"            Label="{Binding Path=ContainerImageLabelText, Mode=OneWay}"            IsRequired="true">            <TextBox Text="{Binding Path=ContainerImage, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />        
</validatable:ValidatableInputField>

**Comment by beccamc** [comment]
Build available here - https://devdiv.visualstudio.com/_apis/resources/Containers/8663755/VSSetup_Release17?itemPath=VSSetup_Release17%2FMicrosoft.ML.ModelBuilder.Gallery.Dev17.vsix

**Comment by NiharikaTipirishetty** [comment]
Issue is fixed, a watermark text is given for all required fields. Hence closing this bug.

![1579_fixed](https://user-images.githubusercontent.com/83570638/133433948-a1906aef-a93a-4e0a-bfa7-63e44bfec145.png)

**Comment by VijayaBharathyA** [comment]
#Closed;
